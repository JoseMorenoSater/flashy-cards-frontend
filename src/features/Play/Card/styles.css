.card-container {
  height: 50%;
  background-color: #c4ebfb;
  text-align: center;
  vertical-align: middle;
}

@media only screen and (max-height: 400px) {
  .card-container {
    height: 80%;
  }
}
