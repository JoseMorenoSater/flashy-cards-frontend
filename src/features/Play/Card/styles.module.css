.cardContainer {
  height: 20em;
  background-color: #c4ebfb;
  text-align: center;
  vertical-align: middle;
  padding: 0.3em;
}

@media only screen and (max-height: 400px) {
  .cardContainer {
    max-height: 80%;
  }
}

.cardImage {
  max-height: 80%;
}

.cardTextContainer {
  height: 20%;
  background-color: bisque;
  padding-top: 0.5em;
  position: absolute;
  bottom: 0;
  width: 100%;
}
