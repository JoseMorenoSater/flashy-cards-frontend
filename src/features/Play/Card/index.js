/**
 *
 * Card
 *
 */

import React from "react";
import PropTypes from "prop-types";
import "./styles.css";

const Card = ({ cardText, flipCard, cardMedia, cardAudio }) => {
  let media;
  if (cardMedia !== undefined) {
    switch (cardMedia.type) {
      case "IMAGE":
        media = (
          <img alt={cardText} src={cardMedia.source} className="img-fluid" />
        );
        break;
      default:
        media = undefined;
    }
  }
  return (
    <div
      className="card-container"
      onClick={() => {
        let audio = new Audio(cardAudio);
        audio.play();
        flipCard();
      }}
    >
      {cardText}
      {media}
    </div>
  );
};
Card.propTypes = {
  cardText: PropTypes.string.isRequired,
  flipCard: PropTypes.func.isRequired,
  cardMedia: PropTypes.object,
  cardAudio: PropTypes.string
};
export default Card;
