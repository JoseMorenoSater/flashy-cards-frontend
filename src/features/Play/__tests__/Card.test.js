import React from "react";
import { create } from "react-test-renderer";
import Card from "../Card";

describe("Card Component", () => {
  it("should render correctly", () => {
    const CardComponent = create(
      <Card cardText={"test"} flipCard={() => {}} />
    ).toJSON();
    expect(CardComponent).toMatchSnapshot();
  });
});
