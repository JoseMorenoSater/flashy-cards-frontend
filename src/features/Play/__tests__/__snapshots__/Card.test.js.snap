// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Component should render correctly 1`] = `
<div
  className="card-container"
  onClick={[Function]}
>
  test
</div>
`;
